### ISSUE TITLE
`SwiftModuleBuildDescription.emitCommandLine` should not always generate output file map

### ISSUE THREAD

**Issue by ahoppen** [swiftlang/swift-package-manager#8038]
`SwiftModuleBuildDescription.emitCommandLine` always writes a new output file map.

https://github.com/swiftlang/swift-package-manager/blob/80b1e179bef19e6eb71b0d20b2cb471053ae1b2c/Sources/Build/BuildDescription/SwiftModuleBuildDescription.swift#L745-L746

This is causing issues in SourceKit-LSP on Windows because we might be running a build (eg. for target preparation), which writes an output file map and at the same time try to get build settings for that file inside SourceKit-LSP, which calls into `emitCommandLine`, also trying to write an output file map, which fails because the output file map is exclusively opened by the build and thus cannot also write the output file map. 

While this might seem like a more theoretical race condition, this caused multiple tests to fail when running SourceKit-LSP tests locally on my machine.

**Comment by rauhul** [comment]
Write output file map should be part of the build manifest instead right?

**Comment by neonichu** [comment]
> Write output file map should be part of the build manifest instead right?

No, that's how it used to be, but that causes several issues. The simplest one being if someone deletes the built artifacts but not the manifest (which is a fine thing to do for a user), builds will fail.

**Comment by dschaefer2** [comment]
Ran into this little surprise working on the Windows linking fixes. The side effect is horrible. Would be nice to fix this properly, just not sure what that would be.

**Comment by rauhul** [comment]
> No, that's how it used to be, but that causes several issues. The simplest one being if someone deletes the built artifacts but not the manifest (which is a fine thing to do for a user), builds will fail.

Maybe I misspoke, I think I mean the build graph

**Comment by dschaefer2** [comment]
Agreed. I was expecting this to be in the build graph just like sources and link file list.

**Comment by ahoppen** [comment]
Just adding here that during SourceKit-LSP PR testing, we saw the following error while calling `BuildDescription.load`.

```
You don't have permission."UserInfo={NSFilePath=C:\Users\swift-ci\jenkins\workspace\sourcekit-lsp-PR-windows\build\tmp\lsp-test\7EB7D3B3\PackageA\.build\index-build\x86_64-unknown-windows-msvc\debug\MyExec.build\output-file-map.json, NSURL=file:///C:/Users/swift-ci/jenkins/workspace/sourcekit-lsp-PR-windows/build/tmp/lsp-test/7EB7D3B3/PackageA/.build/index-build/x86_64-unknown-windows-msvc/debug/MyExec.build/output-file-map.json
```

I double checked that SourceKit-LSP is not performing any other SwiftPM-related operations while it is calling `BuildDescription.load`, so this must be some kind of race condition within SwiftPM. I expect that this failure kind will be fixed when we eliminate this side effect.

[Build log of failing CI run](https://github.com/user-attachments/files/18968925/sourcekit-lsp-PR-windows-2717.txt)
